Kitodo-API/src/main/java/org/kitodo/api/ugh/DocStructInterface.java

```diff
@@ -0,0 +1,628 @@
+     * @param direction
+     *            can be "{@code to}" or "{@code from}". String seems to be
+     *            always "{@code to}".
+     * @return incoming or outgoing {@code Reference}s
+     */
+    List<ReferenceInterface> getAllReferences(String direction);
+
+    /**
+     * Returns all references that are directed from this instance to another.
+     * This are all {@code Reference}s in which this instance is the source.
+     *
+     * @return all outgoing {@code Reference}s
+     */
+    Collection<ReferenceInterface> getAllToReferences();
+
+    /**
+     * Returns all references that are directed from this instance to another
+     * and have a given type. For example, the type "{@code logical_physical}"
+     * refers to references from logical structures to physical structures.
+     *
+     * @param type
+     *            type of the references to return, always "logical_physical"?
+     * @return all outgoing {@code Reference}s of the given type. The return
+     *         type would be sufficient to be an Iterable, but there is a check
+     *         for size()=0.
+     */
+    Collection<ReferenceInterface> getAllToReferences(String type);
+
+    /**
+     * Returns all meta-data for this instance that shall be displayed. This
+     * excludes all {@code Metadata} whose {@code MetadataType} starts with the
+     * {@code HIDDEN_METADATA_CHAR}.
+     *
+     * <p>
+     * Method usage: is null/is non-null -check only.
+     *
+     * @return all meta-data that shall be displayed
+     */
+    Object getAllVisibleMetadata();
+
+    /**
+     * Retrieves the name of the anchor structure, if any, or null otherwise.
+     * Anchors are a special type of document structure, which group other
+     * structure entities together, but have no own content. Imagine a
+     * periodical as such an anchor. The periodical itself is a virtual
+     * structure entity without any own content, but groups all years of
+     * appearance together. Years may be anchors again for volumes, etc.
+     *
+     * @return String, which is null, if it cannot be used as an anchor
+     */
+    String getAnchorClass();
+
+    /**
+     * The function getChild() returns a child of a DocStruct, identified by its
+     * type and an identifier in a meta data field of choice. More formally,
+     * returns the first child matching the given conditions and does not work
+     * recursively. If no matching child is found, throws
+     * NoSuchElementException.
+     *
+     * @param type
+     *            structural type of the child to locate
+     * @param identifierField
+     *            meta data field that holds the identifier to locate the child
+     * @param identifier
+     *            identifier of the child to locate
+     * @return the child, if found
+     * @throws NoSuchElementException
+     *             if no matching child is found
+     */
+    DocStructInterface getChild(String type, String identifierField, String identifier);
```

Comment:
Please adjust the JavaDoc or method signature as they are different.

Reply:
`NoSuchElementException` is a runtime exception which does [not have to be declared](https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html). However, it should be documented here as the using code handles it (unlike other places, where the absence of an object is indicated by returning `null`).

Reply:
Acceptable but I hate any kind of unchecked exceptions. They are misused in many places and are for me a sign of bad coding.